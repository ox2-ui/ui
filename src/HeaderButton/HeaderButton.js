import PropTypes from 'prop-types';
import React from 'react';
import Button from '@ox2/button/Button';

const styles = {
  button: {
    flex: '0 0 auto',
  },
};

/**
 * HeaderButton Component
 */
const HeaderButton = ({
  style,
  onClick,
  label,
  className,
}) => {
  const mergedStyles = {
    ...styles.button,
    ...style,
  };

  return (
    <Button
      btn="header-button small raised"
      className={className}
      onClick={onClick}
      style={mergedStyles}
    >
      {label}
    </Button>
  );
};

HeaderButton.defaultProps = {
  className: '',
  style: {},
};

HeaderButton.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  style: PropTypes.object,
};

export default HeaderButton;
