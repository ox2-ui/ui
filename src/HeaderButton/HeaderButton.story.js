import React from 'react';
import { storiesOf } from '@kadira/storybook';
import HeaderButton from './HeaderButton';

const handleOnClick = () => console.log('🍇'); // eslint-disable-line no-console

storiesOf('HeaderButton', module)
  .add('default', () => (
    <HeaderButton onClick={handleOnClick} label="Add" />
  ));
