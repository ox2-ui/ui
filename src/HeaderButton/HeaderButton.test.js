/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import HeaderButton from './HeaderButton';

const handleOnClick = () => console.log('🍇'); // eslint-disable-line no-console

it('renders correctly', () => {
  const wrapper = mount(
    <HeaderButton onClick={handleOnClick} label="Add" />
  );

  expect(wrapper).toMatchSnapshot();
});
