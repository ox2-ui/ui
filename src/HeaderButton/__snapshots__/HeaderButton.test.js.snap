// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<HeaderButton
  className=""
  label="Add"
  onClick={[Function]}
  style={Object {}}
>
  <Button
    btn="header-button small raised"
    className=""
    onClick={[Function]}
    style={
      Object {
        "flex": "0 0 auto",
      }
    }
  >
    <button
      className=""
      data-btn="header-button small raised"
      onClick={[Function]}
      style={
        Object {
          "flex": "0 0 auto",
        }
      }
    >
      Add
    </button>
  </Button>
</HeaderButton>
`;
